diff -Naur sundials-2.0.2-ORIGINAL/ida/README sundials-2.0.2-PATCHED/ida/README
--- sundials-2.0.2-ORIGINAL/ida/README	2005-03-18 17:11:27.000000000 -0800
+++ sundials-2.0.2-PATCHED/ida/README	2006-06-16 15:04:32.000000000 -0700
@@ -1,5 +1,5 @@
                                IDA
-                    Release 2.2.2, March 2005
+                    Release 2.2.2-1, March 2005
              Alan C. Hindmarsh, Allan G. Taylor, Radu Serban 
               Center for Applied Scientific Computing, LLNL
 
diff -Naur sundials-2.0.2-ORIGINAL/ida/source/idaband.c sundials-2.0.2-PATCHED/ida/source/idaband.c
--- sundials-2.0.2-ORIGINAL/ida/source/idaband.c	2005-01-24 13:43:39.000000000 -0800
+++ sundials-2.0.2-PATCHED/ida/source/idaband.c	2006-06-16 15:06:03.000000000 -0700
@@ -535,7 +535,7 @@
         adjustments using ypj and ewtj if this is small, and a further
         adjustment to give it the same sign as hh*ypj. */
 
-        inc = srur*MAX(ABS(yj), MAX( ABS(hh*ypj), ONE/ewtj));
+        inc = MAX( srur * MAX( ABS(yj), ABS(hh*ypj) ) , ONE/ewtj );
         if (hh*ypj < ZERO) inc = -inc;
         inc = (yj + inc) - yj;
 
@@ -572,7 +572,7 @@
       
       /* Set increment inc exactly as above. */
       
-      inc = srur*MAX(ABS(yj),MAX( ABS(hh*ypj), ONE/ewtj));
+      inc = MAX( srur * MAX( ABS(yj), ABS(hh*ypj) ) , ONE/ewtj );
       if (hh*ypj < ZERO) inc = -inc;
       inc = (yj + inc) - yj;
       if (constraints != NULL) {
diff -Naur sundials-2.0.2-ORIGINAL/ida/source/idadense.c sundials-2.0.2-PATCHED/ida/source/idadense.c
--- sundials-2.0.2-ORIGINAL/ida/source/idadense.c	2004-11-16 14:09:08.000000000 -0800
+++ sundials-2.0.2-PATCHED/ida/source/idadense.c	2006-06-16 15:06:37.000000000 -0700
@@ -497,7 +497,7 @@
     adjustments using yp_j and ewt_j if this is small, and a further
     adjustment to give it the same sign as hh*yp_j. */
 
-    inc = srur*MAX(ABS(yj),MAX( ABS(hh*ypj), ONE/ewt_data[j]));
+    inc = MAX( srur * MAX( ABS(yj), ABS(hh*ypj) ) , ONE/ewt_data[j] );
     if (hh*ypj < ZERO) inc = -inc;
     inc = (yj + inc) - yj;
 
